src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_skadi.cpp

```diff
@@ -274,192 +230,429 @@ class boss_skadi : public CreatureScript
-                                m_uiMovementTimer = 5000;
-                                break;
-                            case 1:
-                                me->GetMotionMaster()->MovePoint(0, Location[2].GetPositionX(), Location[2].GetPositionY(), Location[2].GetPositionZ());
-                                m_uiMovementTimer = 2000;
-                                break;
-                            case 2:
-                                me->GetMotionMaster()->MovePoint(0, Location[3].GetPositionX(), Location[3].GetPositionY(), Location[3].GetPositionZ());
-                                m_uiMovementTimer = 15000;
-                                break;
-                            case 3:
-                                me->GetMotionMaster()->MovePoint(0, Location[69].GetPositionX(), Location[69].GetPositionY(), Location[69].GetPositionZ());
-                                Talk(SAY_DRAKE_BREATH);
-                                Talk(EMOTE_BREATH);
-                                m_uiMovementTimer = 2500;
-                                break;
-                            case 4:
-                                me->GetMotionMaster()->MovePoint(0, Location[70].GetPositionX(), Location[70].GetPositionY(), Location[70].GetPositionZ());
-                                m_uiMovementTimer = 2000;
-                                SpawnTrigger();
-                                break;
-                            case 5:
-                                me->GetMotionMaster()->MovePoint(0, Location[71].GetPositionX(), Location[71].GetPositionY(), Location[71].GetPositionZ());
-                                m_uiMovementTimer = 3000;
-                                break;
-                            case 6:
-                                me->GetMotionMaster()->MovePoint(0, Location[3].GetPositionX(), Location[3].GetPositionY(), Location[3].GetPositionZ());
-                                m_uiWaypointId = 2;
-                                m_uiMovementTimer = 15000;
-                                break;
-                        }
-                        m_uiWaypointId++;
-                    } else m_uiMovementTimer -= diff;
+                        case EVENT_FLYING:
+                            if (Creature* grauf = ObjectAccessor::GetCreature(*me, m_uiGraufGUID))
+                                grauf->AI()->DoAction(DATA_FLYING);
+                            break;
+                        case EVENT_SUMMON:
+                            SpawnMobs();
+                            events.ScheduleEvent(EVENT_SUMMON, 32000);
+                            break;
+                        case EVENT_SPAWN_MOVE:
+                            if (!spawns.empty())
+                            {
+                                for(std::list<Creature*>::const_iterator itr = spawns.begin(); itr != spawns.end(); ++itr)
```

Comment:
there are still Creature* pointers even if I wrote at https://github.com/TrinityCore/TrinityCore/pull/14866#discussion_r32251467 about it.